src/main/cross_game.py

```diff
@@ -78,7 +82,7 @@ def _play(self):
     def apply_action(self, col_index):
         if col_index >= self._nb_columns or col_index < 0:
             raise OutOfGridError(self.current_token_id, col_index, self._nb_columns)
-        if self._grid[col_index][self._nb_rows - 1] != 0:
+        if self.is_legal_action(self._grid, col_index):
```

Comment:
ça devrait être `is_not_legal_action` si on envoie une `ColumnIsFullError` dans le cas où c'est `True`

Reply:
if not is_legal_action